Sim_To_DuT_Interface/DuTLogger/DuTLogger.cpp

```diff
@@ -0,0 +1,339 @@
+ * select for which type of logger you want to change the level.
+ * @param type the type of logger
+ * @param level new level for logging
+ */
+void DuTLogger::changeLogLevel(LOG_LEVEL_CHANGE_ON type, LOG_LEVEL level) {
+    quill::Handler* handler = DuTLogger::getHandlerType(type);
+
+    switch (level) {
+        case LOG_LEVEL::NONE:
+            handler->set_log_level(quill::LogLevel::None);
+            break;
+
+        case LOG_LEVEL::DEBUG:
+            handler->set_log_level(quill::LogLevel::Debug);
+            break;
+
+        case LOG_LEVEL::INFO:
+            handler->set_log_level(quill::LogLevel::Info);
+            break;
+
+        case LOG_LEVEL::WARNING:
+            handler->set_log_level(quill::LogLevel::Warning);
+            break;
+
+        case LOG_LEVEL::ERROR:
+            handler->set_log_level(quill::LogLevel::Error);
+            break;
+
+        case LOG_LEVEL::CRITICAL:
+            handler->set_log_level(quill::LogLevel::Critical);
+            break;
+
+        default:
+            logMessage("Can not change LogLevel! Invalid argument!", LOG_LEVEL::ERROR, true);
+            handler->set_log_level(quill::LogLevel::TraceL3);
+            break;
+    }
+}
+
+/**
+ * Returns the right quill handler for the given type of logger
+ * @param type type of logger
+ * @return connected handler
+ */
+quill::Handler* DuTLogger::getHandlerType(LOG_LEVEL_CHANGE_ON type) {
+    quill::Handler* handler;
+
+    switch (type) {
+        case LOG_LEVEL_CHANGE_ON::CONSOLE_LOG:
+            handler = consoleHandler;
+            break;
+
+        case LOG_LEVEL_CHANGE_ON::FILE_LOG:
+            handler = consoleFileHandler;
+            break;
+
+        default:
+            throw std::invalid_argument("Internal Error! Unknown Typ of <LOG_LEVEL_CHANGED_ON> appeared.");
+    }
+
+    return handler;
+}
+
+/**
+ * Logs the message with consideration of the log level. Please notice that this function will also log the message
+ * into a logfile if the file logger accepts the log level. If you explicitly don't want to log this message into the
+ * logfile, please use another function.
+ * @param msg message that should be logged
+ * @param level the logging level for this message
+ */
+void DuTLogger::logMessage(std::string msg, LOG_LEVEL level) {
+    // hand the message with the right level to the quill framework
+    // the user didn't choose if it has to be logged in the logfile
+    // -> try to log it with the level to the file, the level on the fileLogger will decide if the msg will be written
+    logWithLevel(consoleFileLogger, msg, level);
+}
+
+/**
+ * Logs the message with consideration of the log level. This function gives you additionally the possibility
+ * to choose with the 3rd parameter if you want to log the message into the logfile or explicitly not.
+ * Please notice that there will be no file logging if the logger won't support the log level, even if you
+ * parse 'true' on the 3rd argument.
```

Comment:
typo: pass

Reply:
Is this really a good idea? If I want to explicitly write to file and pass it here why cant I?